src/instructions.js

```diff
@@ -4841,24 +4841,26 @@ t[0xF4] = cpu => {
 
         let i = (cpu.modrm_byte >> 3 & 7) << 2;
 
-        let result = cpu.do_mul32(destination[0] , source[0]);
-        cpu.reg_xmm32s[i] = result[0];
-        cpu.reg_xmm32s[i + 1] = result[1];
+        for (let j = 0; j <= 2; j += 2)
+        {
+            let d = destination[j];
+            let s = source[j];
+            let x = i + j;
 
-        result = cpu.do_mul32(destination[2] , source[2]);
-        cpu.reg_xmm32s[i + 2] = result[0];
-        cpu.reg_xmm32s[i + 3] = result[1];
+            cpu.reg_xmm32s[x] = v86util.imul(d, s);
+            cpu.reg_xmm32s[x + 1] = (d >>> 0) * (s >>> 0) / 0x100000000;
+        }
```

Comment:
Not a fan of this for loop, could you undo the change?

Reply:
Unrolled.